[AIRFLOW-1470] Implement BashSensor operator

diogoalexandrefranco opened the pull request:
Dear Airflow maintainers,

This PR implements a BashSensor operator, which succeeds once a bash command/script returns 0, and keeps poking otherwise. It allows the user to easily construct dag's that wait for events with custom logic.

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [X] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW-1470) issues and references them in the PR title.


### Description
- [X] Here are some details about my PR, including screenshots of any UI changes:
The implementation of the poke method is very similar to the execute method of the BashOperator.

### Tests
- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
* The class _BashSensorTests_ was added, which can be executed directly with ```nosetests tests.contrib.sensors.test_bash_sensor```


### Commits
- [X] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

andyxhadji commented on airflow/operators/__init__.py:
```diff
@@ -76,6 +76,7 @@
         'TimeDeltaSensor',
         'TimeSensor',
         'WebHdfsSensor',
+        'BashSensor'
```
This should be in the community-contributed operators section - rebase before adding, documentation structure has since changed.

diogoalexandrefranco committed:
[AIRFLOW-1470] Implement BashSensor operator

This sensor succeeds once a bash command/script
returns 0, and keeps poking otherwise. The
implementation is very similar to BashOperator.

Fokko commented on airflow/contrib/sensors/bash_sensor.py:
```diff
@@ -0,0 +1,91 @@
+                    script_location
+                )
+                self.log.info("Running command: %s", bash_command)
+                sp = Popen(
```
Can you add `close_fds=True`?

Fokko reviewed (commented):
One minor thing, apart from that LGTM
